processor/src/host/mod.rs

```diff
@@ -111,16 +124,30 @@ pub trait AsyncHost: BaseHost {
 
     /// Returns MAST forest corresponding to the specified digest, or None if the MAST forest for
     /// this digest could not be found in this [AsyncHost].
-    fn get_mast_forest(
-        &self,
-        node_digest: &Word,
-    ) -> impl Future<Output = Option<Arc<MastForest>>> + Send;
+    fn get_mast_forest(&self, node_digest: &Word) -> Option<Arc<MastForest>>;
```

Comment:
Why is this no longer async?

Reply:
`AsyncHostFuture` ripples this all through `-base` and causes `&'store STORE` to be required to be `Sync`, since we return an `Arc<MastForest>`, which does not hold to be `Sync`. The intent is to do this the next PR/a follow-up but test against `-base` and `-client` first.